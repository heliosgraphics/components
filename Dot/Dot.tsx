import React, { useMemo } from "react";
import styles from "./Dot.module.css";
import { getUUID } from "@heliosgraphics/utils/uuid";
import { getColor } from "../utils/color";
import type { DotProps } from "./Dot.types";

const Dot: React.FC<DotProps> = ({ colorAccent, size = 8, color = "blue" }) => {
	const dotId: string = useMemo(() => getUUID(), []);
	const dotColor: string = getColor(color) as string;
	const dotColorAccent: string = useMemo(
		() => getColor(colorAccent, undefined, dotColor),
		[colorAccent, dotColor],
	) as string;

	const dotStyle = {
		height: size + "px",
		maxHeight: size + "px",
		maxWidth: size + "px",
		width: size + "px",
	};

	return (
		<div className={styles.dot} style={dotStyle}>
			<svg width="100%" height="100%" className={dotColor} viewBox="0 0 20 20">
				<linearGradient id={dotId}>
					<stop stopColor={`var(--${dotColorAccent})`} offset="0%" />
					<stop stopColor={`var(--${dotColor})`} offset="100%" />
				</linearGradient>
				<circle fill={`url(#${dotId})`} cx={10} cy={10} r={10} />
			</svg>
		</div>
	);
};

export default Dot;
