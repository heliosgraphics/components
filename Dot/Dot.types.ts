import type { FUIColors } from "@heliosgraphics/library/types/colors"

export interface DotProps {
  color: FUIColors
  colorAccent?: FUIColors
  size: number
}
