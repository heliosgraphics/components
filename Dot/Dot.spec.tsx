import React from 'react';
import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import Dot from "./Dot";

describe("Dot", () => {
	it("Should match snapshot", () => {
		const dotElement = render(<Dot size={8} color="red" />);
		expect(dotElement).toMatchSnapshot();
	});
});
