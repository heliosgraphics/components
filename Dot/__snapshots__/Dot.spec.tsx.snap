// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dot > Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_dot_f4ab0e"
        style="height: 8px; max-height: 8px; max-width: 8px; width: 8px;"
      >
        <svg
          class="red-500"
          height="100%"
          viewBox="0 0 20 20"
          width="100%"
        >
          <lineargradient
            id="00000000-0000-0000-000000000000"
          >
            <stop
              offset="0%"
              stop-color="var(--red-500)"
            />
            <stop
              offset="100%"
              stop-color="var(--red-500)"
            />
          </lineargradient>
          <circle
            cx="10"
            cy="10"
            fill="url(#00000000-0000-0000-000000000000)"
            r="10"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_dot_f4ab0e"
      style="height: 8px; max-height: 8px; max-width: 8px; width: 8px;"
    >
      <svg
        class="red-500"
        height="100%"
        viewBox="0 0 20 20"
        width="100%"
      >
        <lineargradient
          id="00000000-0000-0000-000000000000"
        >
          <stop
            offset="0%"
            stop-color="var(--red-500)"
          />
          <stop
            offset="100%"
            stop-color="var(--red-500)"
          />
        </lineargradient>
        <circle
          cx="10"
          cy="10"
          fill="url(#00000000-0000-0000-000000000000)"
          r="10"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
